internal/run/convert.go

```diff
@@ -22,6 +27,78 @@ func v3ReleaseFound(release string, cfg *action.Configuration) bool {
 	return false
 }
 
+// clientsetFromFile returns a ready-to-use client from a kubeconfig file
+func clientsetFromFile(path string) (*kubernetes.Clientset, error) {
+	config, err := clientcmd.LoadFromFile(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load admin kubeconfig")
+	}
+
+	overrides := clientcmd.ConfigOverrides{Timeout: "15s"}
+	clientConfig, err := clientcmd.NewDefaultClientConfig(*config, &overrides).ClientConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create API client configuration from kubeconfig")
+	}
+
+	client, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create API client")
+	}
+	return client, nil
+}
+
+// getReleaseConfigmaps returns a list of configmaps that are helm v2 releases
+func getReleaseConfigmaps(clientset kubernetes.Interface, release, tillerNamespace, tillerLabel string) (*corev1.ConfigMapList, error) {
+
+	if tillerNamespace == "" {
+		tillerNamespace = "kube-system"
+	}
+	if tillerLabel == "" {
+		tillerLabel = "OWNER=TILLER"
+	}
+	if release != "" {
+		tillerLabel += fmt.Sprintf(",NAME=%s", release)
+	}
+
+	configMaps, err := clientset.CoreV1().ConfigMaps(tillerNamespace).List(metav1.ListOptions{
+		LabelSelector: tillerLabel,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return configMaps, nil
+}
+
+// preserveV2 keeps the helm v2 configmaps by modifying a label
+func preserveV2(clientset kubernetes.Interface, o convertcmd.ConvertOptions) error {
+
+	configMaps, err := getReleaseConfigmaps(clientset, o.ReleaseName, o.TillerNamespace, o.TillerLabel)
+	if err != nil {
+		return err
+	}
+
+	cmClient := clientset.CoreV1().ConfigMaps(o.TillerNamespace)
+
+	var updateErrors []error
+
+	log.Printf("Preserving release versions of %s", o.ReleaseName)
+	for _, item := range configMaps.Items {
+		item.Labels["OWNER"] = "none"
+
+		_, updateErr := cmClient.Update(&item)
+		if updateErr != nil {
+			updateErrors = append(updateErrors, updateErr)
+		}
+	}
+
+	if len(updateErrors) > 0 {
+		return fmt.Errorf("Failure preserving release info of %s", o.ReleaseName)
+	}
```

Comment:
`updateErrors` is never returned.  If it is not necessary to catch the internal errors of the Update client, then simply using a boolean to flag that any error occurred.

Reply:
true, will do that

Reply:
now thinking about it. There might be cases where the update to one of the configmaps fails and others succeed, therefore ending in an incomplete state. It is probably better to fail if there's at most one error.

Reply:
I believe that is current state.  Unless you want to bubble up an error from one of those failed updates, using a boolean means you will not to keep an error array.

Reply:
I changed the code to return as soon as an error is found